code/modules/projectiles/guns/ballistic/rifle.dm

```diff
@@ -113,6 +113,12 @@
 		item_state = "illestren_factory_sawn"
 		mob_overlay_state = item_state
 
+/obj/item/gun/ballistic/rifle/illestren/sawn
```

Comment:
My current understanding is this would have the wrong stats of a sawn off
I think there should probably be a check on init to update the stats if its set to sawn_off?
Code for currently sawoff
```
/obj/item/gun/ballistic/proc/sawoff(mob/user, obj/item/saw)
	if(!saw.get_sharpness() || !is_type_in_typecache(saw, GLOB.gun_saw_types) && !saw.tool_behaviour == TOOL_SAW) //needs to be sharp. Otherwise turned off eswords can cut this.
		return
	if(sawn_off)
		to_chat(user, "<span class='warning'>\The [src] is already shortened!</span>")
		return
	user.changeNext_move(CLICK_CD_MELEE)
	user.visible_message("<span class='notice'>[user] begins to shorten \the [src].</span>", "<span class='notice'>You begin to shorten \the [src]...</span>")

	//if there's any live ammo inside the gun, makes it go off
	if(blow_up(user))
		user.visible_message("<span class='danger'>\The [src] goes off!</span>", "<span class='danger'>\The [src] goes off in your face!</span>")
		return

	if(do_after(user, 30, target = src))
		if(sawn_off)
			return
		user.visible_message("<span class='notice'>[user] shortens \the [src]!</span>", "<span class='notice'>You shorten \the [src].</span>")
		name = "sawn-off [src.name]"
		desc = sawn_desc
		w_class = WEIGHT_CLASS_NORMAL
		item_state = "gun"
		slot_flags &= ~ITEM_SLOT_BACK	//you can't sling it on your back
		slot_flags |= ITEM_SLOT_BELT		//but you can wear it on your belt (poorly concealed under a trenchcoat, ideally)
		recoil = SAWN_OFF_RECOIL
		sawn_off = TRUE
		update_appearance()
		return TRUE
```
this effect should still occur tho 
```
/obj/item/gun/ballistic/process_fire(atom/target, mob/living/user, message = TRUE, params = null, zone_override = "", bonus_spread = 0)
	if (sawn_off)
		bonus_spread += SAWN_OFF_ACC_PENALTY
	. = ..()
```

Reply:
This will simplify adding spawned in sawed offs to just the sawn_off = TRUE since the name and everything else will get updated

keep in mind the current sawoff proc returns if the gun already has sawn_off = TRUE. You should probally have sawoff handle just the action while you add a on_sawoff to update stats. 
/obj/item/gun/ballistic/shotgun/doublebarrel/improvised/sawn has the same issue

Reply:
oh and illestern sawoff proc
```
/obj/item/gun/ballistic/rifle/illestren/sawoff(mob/user)
	. = ..()
	if(.)
		spread = 36
		spread_unwielded = 108
		item_state = "illestren_sawn"
		mob_overlay_state = item_state
		weapon_weight = WEAPON_MEDIUM //you can fire it onehanded, makes it worse than worse than useless onehanded, but you can

```
unrelated but this is meant to be illestern/factory/sawoff
```
/obj/item/gun/ballistic/rifle/illestren/sawoff(mob/user)
	. = ..()
	if(.)
		item_state = "illestren_factory_sawn"
		mob_overlay_state = item_state
```

Reply:
I can do this as a seperate pr if you dont want to do this just ping me